core/templates/dev/head/services/contextual/UrlServiceSpec.ts

```diff
@@ -208,6 +208,36 @@ describe('Url Service', function() {
     ).toBe(null);
   });
 
+  it('should correctly retrieve collection id from url in exploration player',
+    function() {
+      mockLocation.search = '?collection_id=abcdefghijkl';
+      expect(
+        UrlService.getCollectionIdFromExplorationUrl()
+      ).toBe('abcdefghijkl');
+
+      mockLocation.search = '?collection=abcdefghijkl';
+      expect(
+        UrlService.getCollectionIdFromExplorationUrl()
+      ).toBe(null);
+
+      mockLocation.search = '?collection_id=abcdefghijkl&parent=mnopqrst';
+      expect(
+        UrlService.getCollectionIdFromExplorationUrl()
+      ).toBe(null);
+    }
+  );
+
+  it('should correctly retrieve exploration version from the url', function() {
+    mockLocation.search = '?v=1';
+    expect(UrlService.getExplorationVersionFromUrl()).toBe(1);
+
+    mockLocation.search = '?v=1#version=0.0.9';
+    expect(UrlService.getExplorationVersionFromUrl()).toBe(1);
+
+    mockLocation.search = '?another=1';
+    expect(UrlService.getExplorationVersionFromUrl()).toBe(null);
+  });
```

Comment:
Add test for cases when `v` is second parameter.

Reply:
Done